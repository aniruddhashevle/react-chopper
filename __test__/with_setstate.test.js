import React from "react";
import renderer from "react-test-renderer";

import { shallow, mount, render } from "enzyme";

class WithSetState extends React.Component {
  constructor(props) {
    super(props);
    this.state = { firstname: "chopper" };
  }

  render() {
    return <div>Hello react {this.state.firstname}</div>;
  }

  componentDidMount() {
    this.setState({ firstname: "chopper changed kk" });
  }
}

it("renders component correctly", () => {
  const tree = renderer.create(<WithSetState />).toJSON();
  expect(tree).toMatchSnapshot();
});
