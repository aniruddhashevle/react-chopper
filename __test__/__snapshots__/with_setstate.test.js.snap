// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<div>
  Hello react 
  chopper changed
</div>
`;
